### DISCUSSION TITLE
1.88 error: "'DC': undeclared identifier" in imgui.cpp

### DISCUSSION THREAD

**Discussion by ItsNovaAndCrap** [ocornut/imgui#6195] [New Users Build/Link/Run issues ONLY!]
Whenever I try to build, it says that DC isn't declared! I am using TheCherno's fork of ImGui, I would make an issue over there but it doesn't have an issues or discussions page! I am using ImGui 1.88.

**Comment by ocornut** [comment]
Sorry this doesn't constitute a viable support request, you are providing zero context.

**Comment by ItsNovaAndCrap** [comment]
sry, i am trying to create a game engine with thecherno's tutorial. I currently just started implementing imgui and got that error when I tried to build.

**Reply by PathogenDavid** [reply]
This doesn't really add any details that couldn't already be inferred from your main question. Before we can even guess what might be wrong with your environment we need way more detail.

Ideally you should be providing full error message, line numbers, the contents of that line (or the specific Git commit you're on or a permalink to the line on GitHub's web interface), what platform you're building for, and what compiler+build system you're using.

**Comment by ItsNovaAndCrap** [comment]
Gotcha

I am building for x64 platforms.
Here's the full error and the lines that this error is coming from:
```
Severity Code	Description	                Project	File	                                Line	Suppression State
Error	C2065	'DC': undeclared identifier	ImGui	D:\dev\Mist\Mist\vendor\ImGui\imgui.cpp	3839	

Severity Code	Description	                Project	File	                                Line	Suppression State
Error	C2065	'DC': undeclared identifier	ImGui	D:\dev\Mist\Mist\vendor\ImGui\imgui.cpp	3841	
```
and here are the lines that the errors are coming from:
```cpp
3839 |     for (int i = 0; i < DC.Layouts.Data.Size; i++)

3841 |        ImGuiLayout* layout = (ImGuiLayout*)DC.Layouts.Data[i].val_p;
```
as for the build system, i'm using the default one that comes with visual studio 2022 (msvc)

**Reply by PathogenDavid** [reply]
These errors are coming from changes made to the fork of Dear ImGui that you're using. You're going to have to take it up with @TheCherno.

Compare `ImGui::SetActiveID` between [vanilla Dear ImGui `docking`](https://github.com/ocornut/imgui/blob/e25e4526cd41cd6536194de098540d54244f54e9/imgui.cpp#L3819-L3823) with [TheCherno's Dear ImGui fork](https://github.com/TheCherno/imgui/blob/781a4ffc674d98dfd2b4d42747e1cd27887fac36/imgui.cpp#L3834-L3844).

**Reply by VolcanicGameStudios** [reply]
Found out whats wrong! before the DC you need to put window->

So the end result looks like this

for (int i = 0; i < window->DC.Layouts.Data.Size; i++)

       ImGuiLayout* layout = (ImGuiLayout*)window->DC.Layouts.Data[i].val_p;

**Reply by bluueyes** [reply]
thanks
